src/Compilers/Core/Portable/InternalUtilities/ImmutableArrayExtensions.cs

```diff
@@ -9,24 +9,10 @@ namespace Roslyn.Utilities
     internal static class ImmutableArrayExtensions
     {
         internal static ImmutableArray<T> ToImmutableArrayOrEmpty<T>(this IEnumerable<T> items)
-        {
-            if (items == null)
-            {
-                return ImmutableArray.Create<T>();
-            }
-
-            return ImmutableArray.CreateRange<T>(items);
-        }
+            => items == null ? ImmutableArray<T>.Empty : ImmutableArray.CreateRange(items);
 
         internal static ImmutableArray<T> ToImmutableArrayOrEmpty<T>(this ImmutableArray<T> items)
-        {
-            if (items.IsDefault)
-            {
-                return ImmutableArray.Create<T>();
-            }
-
-            return items;
-        }
+            => items.IsDefault ? ImmutableArray<T>.Empty : ImmutableArray.CreateRange(items);
```

Comment:
I think this still boxes `items` to `IEnumerable<T>`, I don't see an overload `ImmutableArray.CreateRange(ImmutableArray<T>)`. Probably want to just return the array as-is: `items.IsDefault ? ImmutableArray<T>.Empty : items`